app/assets/javascripts/3dbio_viewer/src/webapp/components/viewer-selector/viewer-selector.hooks.ts

```diff
@@ -62,10 +64,15 @@ export function useViewerState(section: Section): ViewerState {
             case "uploaded": {
                 const params2 = {
                     ...params,
+                    type: values.type,
                     chain: values.chain,
                     profile: values.profile,
                 } as UploadedParams;
-                const selection = getSelectionFromUploadDataToken(params2.token, params2.chain);
+                const selection = getSelectionFromUploadDataToken(
+                    params2.token,
+                    params2.chain,
+                    params2.type
+                );
```

Comment:
[minor] just a general rule. When we are passing all (or many) fields of an object (here: params2), it seems that they form a unity, and then the function can just take this single object.